src/channel_internal.c

```diff
@@ -392,10 +481,10 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_push, THANDLE(CHA
         /*Codes_SRS_CHANNEL_INTERNAL_43_124: [ If the first operation in the list of pending operations contains a non-NULL pull_callback: ]*/
         else if (CONTAINING_RECORD(channel_internal_ptr->op_list.Flink, CHANNEL_OP, anchor)->pull_callback != NULL)
```

Comment:
>pull_callback [](http://example.com/codeflow?start=93&length=13)

`pull_callback` is evaluated once here, and again in line 309. Probably design flaw.

Revisit design to not evaluate the same condition twice.

Reply:
Suspicion of "same for push_callback"

Reply:
Task #29882176